Lib/test/test_unicode.py

```diff
@@ -17,6 +17,37 @@
 from test import support, string_tests
 from test.support.script_helper import assert_python_failure
 
+# First invalid code point past the valid range
+INVALID_CODEPOINT = sys.maxunicode + 1
+
+# Characters in the ASCII range (U+0000-U+007f)
+ASCII_CHAR = "a"
+ASCII_LAST_CHAR = "\x7f"
+# Characters in the UCS1 ("latin1") range (U+0000-U+00ff),
+# but not in the ASCII range
+UCS1_CHAR = "\xe9"
+UCS1_LAST_CHAR = "\xff"
+# Characters in UCS2 ("BMP") range (U+0000-U+FFFF),
+# but not in the UCS1 range
+UCS2_CHAR = "\u20ac"  # euro sign
+UCS2_LAST_CHAR = "\uffff"
+# Characters in UCS4 ("Astral") range (U+0000-U+FFFF),
+# but not in the UCS2 range
```

Comment:
"Astral" means specifically "not the BMP", so this isn't quite right. (Hm, also it should say "U+10FFFF".)

Perhaps just `# Characters outside the BMP ("astral" characters)`.